addons/account/models/account_move.py

```diff
@@ -5111,7 +5114,10 @@ def _post(self, soft=True):
             if move.state in ['posted', 'cancel']:
                 validation_msgs.add(_('The entry %(name)s (id %(id)s) must be in draft.', name=move.name, id=move.id))
             if not move.line_ids.filtered(lambda line: line.display_type not in ('line_section', 'line_note')):
-                validation_msgs.add(_('You need to add a line before posting.'))
+                raise UserError(_(
+                    "Make sure you add a line before posting the invoice. Invoices can't be created from thin air.\n"
+                    "You'll thank me later for avoiding a weird explanation to your auditor :)"
+                ))
```

Comment:
This doesn't only apply to invoices, but to any journal entry. We probably want to update the message accordingly.